src/Bicep.LangServer/Handlers/BicepGenerateParamsCommandHandler.cs

```diff
@@ -0,0 +1,147 @@
+using Azure.Deployments.Core.Entities;
+using Azure.Deployments.Core.Helpers;
+using Bicep.Core.Analyzers.Linter;
+using Bicep.Core.Configuration;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Emit;
+using Bicep.Core.Features;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Registry;
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.Workspaces;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Utils;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    // This handler is used to generate compiled parameters.json file for given a bicep file path.
+    // It returns generate-params succeeded/failed message, which can be displayed approriately in IDE output window
+    public class BicepGenerateParamsCommandHandler : ExecuteTypedResponseCommandHandlerBase<string, string>
+    {
+        private readonly ICompilationManager compilationManager;
+        private readonly EmitterSettings emitterSettings;
+        private readonly IFeatureProvider features;
+        private readonly IFileResolver fileResolver;
+        private readonly IModuleDispatcher moduleDispatcher;
+        private readonly INamespaceProvider namespaceProvider;
+        private readonly IConfigurationManager configurationManager;
+
+        public BicepGenerateParamsCommandHandler(ICompilationManager compilationManager, ISerializer serializer, IFeatureProvider features, EmitterSettings emitterSettings, INamespaceProvider namespaceProvider, IFileResolver fileResolver, IModuleDispatcher moduleDispatcher, IConfigurationManager configurationManager)
+            : base(LangServerConstants.GenerateParamsCommand, serializer)
+        {
+            this.compilationManager = compilationManager;
+            this.emitterSettings = emitterSettings;
+            this.features = features;
+            this.namespaceProvider = namespaceProvider;
+            this.fileResolver = fileResolver;
+            this.moduleDispatcher = moduleDispatcher;
+            this.configurationManager = configurationManager;
+        }
+
+        public override Task<string> Handle(string bicepFilePath, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(bicepFilePath))
+            {
+                throw new ArgumentException("Invalid input file path");
+            }
+
+            DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepFilePath);
+            string output = GenerateCompiledParametersFileAndReturnOutputMessage(bicepFilePath, documentUri);
+
+            return Task.FromResult(output);
+        }
+
+        private string GenerateCompiledParametersFileAndReturnOutputMessage(string bicepFilePath, DocumentUri documentUri)
+        {
```

Comment:
@ucheNkadiCode Given this bicep file:
```bicep
param nodefault string
@description('hello')
param hasdefault string = 'hello'
```
The params generated is this:
```json
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "metadata": {
    "_generator": {
      "name": "bicep",
      "version": "0.6.68.62354"
    }
  },
  "parameters": {
    "nodefault": {},
    "hasdefault": {
      "value": "hello"
    }
  }
}
```
This writes out the default values from the bicep file into the params file.  I don't think that's the right thing to do - I think by parameters with default values should be left out of the param file so that they actually pick up the default in the bicep file if it changes.

The ARM extension actually asks whether to write out all params or only those without default values:
![image](https://user-images.githubusercontent.com/6913354/171502170-18c5376e-8754-47b7-9dab-88b0128fba35.png)
Either way, it does *not* copy over the default values, e.g.:
Template:
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "nodefault": {
            "type": "string",
            "metadata": {
                "description": "description"
            }
        },
        "hasdefault": {
            "type": "string",
            "defaultValue": "my default value"
        }
    },
    "resources": []
}
```
Generated params file with "all parameters" option:
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "nodefault": {
            "value": "" // TODO: Fill in parameter value
        },
        "hasdefault": {
            "value": "my default value"
        }
    }
}
```
Generated params file with "only required parameters" option:
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "nodefault": {
            "value": "" // TODO: Fill in parameter value
        }
    }
}
```

Reply:
Just as a side note, the ARM extension takes a different approach to updating params files...  It uses a lightbulb:
![image](https://user-images.githubusercontent.com/6913354/171503618-52d6193b-a1ba-486b-b7dc-dc49059e27ee.png)

Reply:
@ucheNkadiCode Let's discuss at sync.

Reply:
TODO: Discuss on Monday

Reply:
Decision for now: Just skip those with expressions as default values for now.

Reply:
as we discussed, if parameter has a default value defined in the bicep file, generate-params command ignore it when building the parameters.json file